diff --git a/Makefile b/Makefile
index 9c7cd5f..be8ab8c 100644
--- a/Makefile
+++ b/Makefile
@@ -0,0 +1,2 @@
+YTKACE_PROJECT_DIR := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
+
@@ -76 +78 @@ YTKACE_CFLAGS += -Wno-module-import-in-extern-c
-YTKACE_CFLAGS += -I$(THEOS_PROJECT_DIR)/Vendor/FFmpeg/include
+YTKACE_CFLAGS += -I"$(YTKACE_PROJECT_DIR)/Vendor/FFmpeg/include"
@@ -81,3 +83,3 @@ YTKACE_LDFLAGS = -Wl,-install_name,@rpath/YTKACE.dylib
-YTKACE_LDFLAGS += $(THEOS_PROJECT_DIR)/Vendor/FFmpeg/lib/libavformat.a
-YTKACE_LDFLAGS += $(THEOS_PROJECT_DIR)/Vendor/FFmpeg/lib/libavcodec.a
-YTKACE_LDFLAGS += $(THEOS_PROJECT_DIR)/Vendor/FFmpeg/lib/libavutil.a
+YTKACE_LDFLAGS += "$(YTKACE_PROJECT_DIR)/Vendor/FFmpeg/lib/libavformat.a"
+YTKACE_LDFLAGS += "$(YTKACE_PROJECT_DIR)/Vendor/FFmpeg/lib/libavcodec.a"
+YTKACE_LDFLAGS += "$(YTKACE_PROJECT_DIR)/Vendor/FFmpeg/lib/libavutil.a"
@@ -89,2 +91,2 @@ after-all::
-	@mkdir -p "$(THEOS_PROJECT_DIR)/dist"
-	@cp "$(THEOS_OBJ_DIR)/YTKACE.dylib" "$(THEOS_PROJECT_DIR)/dist/YTKACE.dylib"
+	@mkdir -p "$(YTKACE_PROJECT_DIR)/dist"
+	@cp "$(THEOS_OBJ_DIR)/YTKACE.dylib" "$(YTKACE_PROJECT_DIR)/dist/YTKACE.dylib"
@@ -94,2 +96,2 @@ after-stage::
-	@cp -R "$(THEOS_PROJECT_DIR)/Resources/YTKACE.bundle" "$(THEOS_STAGING_DIR)/Library/MobileSubstrate/DynamicLibraries/YTKACE.bundle"
-	@cp "$(THEOS_PROJECT_DIR)/YTKACE.plist" "$(THEOS_STAGING_DIR)/Library/MobileSubstrate/DynamicLibraries/YTKACE.plist"
+	@cp -R "$(YTKACE_PROJECT_DIR)/Resources/YTKACE.bundle" "$(THEOS_STAGING_DIR)/Library/MobileSubstrate/DynamicLibraries/YTKACE.bundle"
+	@cp "$(YTKACE_PROJECT_DIR)/YTKACE.plist" "$(THEOS_STAGING_DIR)/Library/MobileSubstrate/DynamicLibraries/YTKACE.plist"
